spec/API_specification/signatures/array_object.py

```diff
@@ -0,0 +1,949 @@
+from ._types import array, dtype, device, Optional, Tuple, Union, Any, PyCapsule, Enum, ellipsis
+
+class array():
+    def __init__(self) -> None:
+        """
+        Initialize the attributes for the array object class.
+        """
+
+    @property
+    def dtype() -> dtype:
+        """
+        Data type of the array elements.
+
+        Returns
+        -------
+        out: dtype
+            array data type.
+        """
+
+    @property
+    def device() -> device:
```

Comment:
This is overriding the name `device` for the `to_device` signature below, because of this:

```py
>>> from typing import List
>>> class Test:
...     @property
...     def List(self) -> List:
...         pass
...
...     def meth(self, arg: List):
...         pass
>>> Test.meth.__annotations__['arg']
<property object at 0x106fea770>
```

So we need to do something like `from ._types import device as Device`. The same thing would also happen with `dtype` except it isn't used anywhere else currently. 

I wonder, though, if we shouldn't be using uppercase for the type names anyway.